gpslogger/src/main/java/com/mendhak/gpslogger/GeneralLocationListener.java

```diff
@@ -69,8 +72,17 @@ public void onLocationChanged(Location loc) {
                 b.putInt(BundleConstants.SATELLITES_FIX, satellitesUsedInFix);
 
                 loc.setExtras(b);
-                loggingService.onLocationChanged(loc);
 
+                if (listenerName.equalsIgnoreCase(BundleConstants.PASSIVE)){
+                    long now = System.currentTimeMillis();
+                    LOG.debug("Passive location listener Interval " + (preferenceHelper.getPassiveFilterInterval() * 1000) + "ms");
+                    if (now - lastUpdateTime >= (preferenceHelper.getPassiveFilterInterval() * 1000)) {
+                        lastUpdateTime = now;
+                        loggingService.onLocationChanged(loc);
+                    }
+                }else{
+                    loggingService.onLocationChanged(loc);
+                }
```

Comment:
There's no need to make changes to GeneralLocationListener, I'd rather it stays thin. Instead, you can let the passive location through and do the filtering in here (where other filtering is also happening): 

https://github.com/mendhak/gpslogger/blob/972c84edf83f28f82c1d2f75f47833be18212b70/gpslogger/src/main/java/com/mendhak/gpslogger/GpsLoggingService.java#L889-L892

Reply:
In here instead of the passive location checking the previous location time, you can also store the last passive location time in a session variable. 

For example for normal locations you can see session.getLatestTimeStamp(), so similarly for passive you can create a session.getLatestPassiveTimeStamp and session.setLatestPassiveTimeStamp after accepting it. 

In that block if a location is not acceptable, just return; else let it continue.